libs/filesystem/cmake/FindCapnp.cmake

```diff
@@ -0,0 +1,197 @@
+#
+# Finds the Cap'n Proto libraries, and compiles schema files.
+#
+# Configuration variables (optional):
+#   CAPNPC_OUTPUT_DIR
+#       Directory to place compiled schema sources (default: the same directory as the schema file).
+#   CAPNPC_IMPORT_DIRS
+#       List of additional include directories for the schema compiler.
+#       (CMAKE_CURRENT_SOURCE_DIR and CAPNP_INCLUDE_DIRS are always included.)
+#   CAPNPC_SRC_PREFIX
+#       Schema file source prefix (default: CMAKE_CURRENT_SOURCE_DIR).
+#   CAPNPC_FLAGS
+#       Additional flags to pass to the schema compiler.
+#
+# Variables that are discovered:
+#   CAPNP_EXECUTABLE
+#       Path to the `capnp` tool (can be set to override).
+#   CAPNPC_CXX_EXECUTABLE
+#       Path to the `capnpc-c++` tool (can be set to override).
+#   CAPNP_INCLUDE_DIRS
+#       Include directories for the library's headers (can be set to override).
+#   CANP_LIBRARIES
+#       The Cap'n Proto library paths.
+#   CAPNP_LIBRARIES_LITE
+#       Paths to only the 'lite' libraries.
+#   CAPNP_DEFINITIONS
+#       Compiler definitions required for building with the library.
+#   CAPNP_FOUND
+#       Set if the libraries have been located.
+#
+# Example usage:
+#
+#   find_package(CapnProto REQUIRED)
+#   include_directories(${CAPNP_INCLUDE_DIRS})
+#   add_definitions(${CAPNP_DEFINITIONS})
+#
+#   capnp_generate_cpp(CAPNP_SRCS CAPNP_HDRS schema.capnp)
+#   add_executable(a a.cc ${CAPNP_SRCS} ${CAPNP_HDRS})
+#   target_link_library(a ${CAPNP_LIBRARIES})
+#
+# For out-of-source builds:
+#
+#   set(CAPNPC_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR})
+#   include_directories(${CAPNPC_OUTPUT_DIR})
+#   capnp_generate_cpp(...)
+#
+
+# CAPNP_GENERATE_CPP ===========================================================
```

Comment:
Why would I delete documentation? Regardless, this is just a file I copied from our repo. I don't intend to modify it